webapp/src/ts/modules/contacts/contacts.component.ts

```diff
@@ -314,6 +314,51 @@ export class ContactsComponent implements OnInit, AfterViewInit, OnDestroy {
     return this.sortDirection === 'last_visited_date';
   }
 
+  private getDocumentIds(contactsList, usersHomePlace): string[] {
+    const docIds = contactsList.map((item) => item._id);
+    if (usersHomePlace && !docIds.includes(usersHomePlace._id)) {
+      docIds.push(usersHomePlace._id);
+    }
+    return docIds;
+  }
+
+  private findHomeIndex(updatedContacts: any[]): number {
+    return _findIndex(updatedContacts, contact => contact._id === this.usersHomePlace._id);
+  }
+
+  private checkAdditionalListItem(homeIndex: number) {
+    this.additionalListItem =
+      !this.filters.search &&
+      (this.additionalListItem || !this.appending) &&
+      homeIndex === -1;
+  }
+
+  private updateUsersHomePlace(updatedContacts: any[], homeIndex: number) {
+    if (homeIndex !== -1) {
+      this.usersHomePlace = updatedContacts[homeIndex];
+
+      if (!this.appending) {
+        updatedContacts.splice(homeIndex, 1);
+        updatedContacts.unshift(this.usersHomePlace);
+      }
+    } else if (this.additionalListItem) {
+      updatedContacts.unshift(this.usersHomePlace);
+    }
+  }
+
+  private setUsersHomePlace(updatedContacts) {
+    if (!this.usersHomePlace) {
+      return updatedContacts;
+    }
+
+    const homeIndex = this.findHomeIndex(updatedContacts);
+
+    this.checkAdditionalListItem(homeIndex);
+    this.updateUsersHomePlace(updatedContacts, homeIndex);
+
+    return updatedContacts;
```

Comment:
Then you can just return here directly

```suggestion
    return this.updateUsersHomePlace(updatedContacts, homeIndex);
```